### PR TITLE
Introduce 3.1 and 4.0 upgrade files

### PR DESCRIPTION
**Pull request by xabbuh**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets |  |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

File: UPGRADE-3.1.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+-------------------
+
+ * Using unsupported configuration keys in YAML configuration files has been deprecated and will raise an exception in Symfony 4.0.
+
+ * Using unsupported options to configure service aliases has been deprecated and will raise an exception in Symfony 4.0.
+
+Form
+----
+
+ * The `choices_as_values` option of the `ChoiceType` has been deprecated and will be removed in Symfony 4.0.
```
Comment (javiereguiluz) [inline UPGRADE-3.1.md]: In this line I miss the *"... use XXX instead"*  Or is it a removed feature without alternative?

Comment (xabbuh) [inline UPGRADE-3.1.md]: There is no alternative. The option only still exists as a remainer of the choices transformation done in Symfony 2.7.

Comment (stof) [inline UPGRADE-3.1.md]: This is a no-op option in 3.x

**Comment by fabpot** [comment]
Can you add line breaks like done in the other UPGRADE files?

**Comment by xabbuh** [comment]
@fabpot Sorry, not sure I understand what you mean.

**Comment by fabpot** [comment]
I mean wrapping at around 80 chars.

**Comment by xabbuh** [comment]
Oh sure, it seems I wrapped around 100 characters instead.

**Comment by javiereguiluz** [comment]
:+1:
Status: Reviewed

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**